internal/capzcredentials/capz.go

```diff
@@ -0,0 +1,61 @@
+package capzcredentials
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetAzureCredentialsFromMetadata(ctx context.Context, ctrlClient client.Client, obj metav1.ObjectMeta) (string, string, string, string, error) {
```

Comment:
what about putting the 4 strings into a struct so you can return 2 things instead of 5?

Reply:
good point, refactored it